.github/workflows/tests.yml

```
@@ -0,0 +1,24 @@
+name: tests
+
+on: [push, pull_request]
+
+jobs:
+  test:
+
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        ruby-version: [2.7, 3.0, 3.1, 3.2]
+
+    container:
+      image: ruby:${{ matrix.ruby-version }}
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Installing ruby dependencies
+        run: bundle check --path=vendor/bundle || bundle install --path=vendor/bundle
+      - name: Run tests
+        run: bundle exec rake
+      - name: Run benchmarks
+        run: bundle exec rake benchmarks
```

```suggestion
    steps:
      - uses: actions/checkout@v3
      - name: Set up Ruby
        uses: ruby/setup-ruby@v1
        with:
          ruby-version: ${{ matrix.ruby-version }}
          bundler-cache: true
      - name: Run tests
        run: bundle exec rake
```
let's consider separate jobs for benchmarks and rubocop(planning to introduce down the line)